tensorflow/core/kernels/non_max_suppression_op_test.cc

```diff
@@ -861,4 +861,474 @@ TEST_F(NonMaxSuppressionWithOverlapsOpTest, TestEmptyInput) {
   test::ExpectTensorEqual<int>(expected, *GetOutput(0));
 }
 
+class CombinedNonMaxSuppressionOpTest : public OpsTestBase {
+ protected:
+  void MakeOp(bool pad_per_class = false) {
+    TF_EXPECT_OK(NodeDefBuilder("combined_non_max_suppression_op",
+                                "CombinedNonMaxSuppression")
+                     .Input(FakeInput(DT_FLOAT))
+                     .Input(FakeInput(DT_FLOAT))
+                     .Input(FakeInput(DT_INT32))
+                     .Input(FakeInput(DT_INT32))
+                     .Input(FakeInput(DT_FLOAT))
+                     .Input(FakeInput(DT_FLOAT))
+                     .Attr("pad_per_class", pad_per_class)
+                     .Finalize(node_def()));
+    TF_EXPECT_OK(InitOp());
+  }
+};
+
+TEST_F(CombinedNonMaxSuppressionOpTest, TestEmptyInput) {
+  MakeOp();
+  AddInputFromArray<float>(TensorShape({0, 0, 0, 4}), {});
+  AddInputFromArray<float>(TensorShape({0, 0, 0}), {});
+  AddInputFromArray<int>(TensorShape({}), {30});
+  AddInputFromArray<int>(TensorShape({}), {10});
+  AddInputFromArray<float>(TensorShape({}), {.5f});
+  AddInputFromArray<float>(TensorShape({}), {0.0f});
+  TF_ASSERT_OK(RunOpKernel());
+
+  //boxes
+  Tensor expected_boxes(allocator(), DT_FLOAT, TensorShape({0, 10, 4}));
+  test::FillValues<float>(&expected_boxes, {});
+  test::ExpectTensorEqual<float>(expected_boxes, *GetOutput(0));
+  
+  //scores
+  Tensor expected_scores(allocator(), DT_FLOAT, TensorShape({0, 10}));
+  test::FillValues<float>(&expected_scores, {});
+  test::ExpectTensorEqual<float>(expected_scores, *GetOutput(1));
+  
+  //classes
+  Tensor expected_classes(allocator(), DT_FLOAT, TensorShape({0, 10}));
+  test::FillValues<float>(&expected_classes, {});
+  test::ExpectTensorEqual<float>(expected_classes, *GetOutput(2));
+  
+  // valid
+  Tensor expected_valid_d(allocator(), DT_INT32, TensorShape({0}));
+  test::FillValues<int>(&expected_valid_d, {});
+  test::ExpectTensorEqual<int>(expected_valid_d, *GetOutput(3));
+  
+  // indices
+  Tensor expected_indices(allocator(), DT_INT32, TensorShape({0, 10}));
+  test::FillValues<int>(&expected_indices, {});
+  test::ExpectTensorEqual<int>(expected_indices, *GetOutput(4));
+}
+
+TEST_F(CombinedNonMaxSuppressionOpTest, TestSelectFromThreeClusters) {
+  MakeOp();
+  AddInputFromArray<float>(
+      TensorShape({1, 6, 1, 4}),
```

Comment:
might be readable to make test data in 6x4 shape?

Reply:
I have some checks in the kernel code which expects batch size to be >=1 so if i set tensor shape to {6,4} it will fail those checks. It also expects boxes to be 4 dimensional in `ParseAndCheckCombinedNMSBoxSizes`